modules/mobile/app/helpers/mobile/list_filter.rb

```diff
@@ -7,88 +7,86 @@ class ListFilter
     class FilterError < StandardError
     end
 
-    PERMITTED_OPERATIONS = %w[eq notEq].freeze
+    PERMITTED_OPERATIONS = %w[eq not_eq].freeze
 
-    def initialize(collection, filter_params)
-      @collection = collection
+    def initialize(list, filter_params)
+      @list = list
       @filter_params = filter_params
     end
 
     # Accepts params:
-    #   @collection - a Common::Collection of Common::Base models
+    #   @list - an array of Common::Resource or Common::Base models
     #   @filter_params - should be an ActionController::Parameters object which should be passed in from the
     #     controller via @params[:filter]. This will pass in another ActionController::Parameters object.
-    # Returns: a new Common::Collection of Common::Base models that match the provided filters
-    def self.matches(collection, filter_params)
-      filterer = new(collection, filter_params)
+    # Returns an array containing:
+    #   an array of Common::Resource or Common::Base models that match the provided filters
+    #   a hash of any errors encountered or nil if no errors were encountered
+    def self.matches(list, filter_params)
+      filterer = new(list, filter_params)
       filterer.result
     end
 
     def result
       validate!
-      metadata = @collection.metadata.merge(filter: filters)
-      Common::Collection.new(data: matches, metadata: metadata, errors: @collection.errors)
+      [matches, nil]
```

Comment:
It's obvious looking at the code and comments here that the nil is where any errors would go but would it be more explicit and consistent if it was always a hash with a key `filter_error` and then the value of it could just be nil? that way anyone implementing it does need to do a nil check before looking for `filter_error`. Though I guess it'd just be as easy as having a Safety operator prefix the check for error

Reply:
Yeah, I wasn't thrilled about switching to a pattern of returning two objects at all, but given that the pattern has changed somewhat and is no longer being merged with (potentially) other errors, it might make more sense to get rid of the hash altogether and just return the error string or nil.

Reply:
Works for me.

Reply:
Ok, I just ditched the hash altogether. Thanks for pointing this out to me because it was really just a leftover from what I was doing before.